cli/001-cli-improve-piping-support.md

```diff
@@ -0,0 +1,33 @@
+- Start Date: 2021-06-11
+- RFC PR:
+- Issue:
+
+### Support better piping of output 
+Problem: currently piping the output into other tools is not well supported. For example utilities such as `grep`, `wc`, `jq` may work with very limited set of commands / in limited functionality. 
+
+Structure outputs so it would be easy to pipe it into common utilities. 
+
+To achieve that:
+
+ - `wc`, `more` support: integrate output with `less` cmd by default if it's available in the OS. If `less` is not found, create and use builtin functiniality to output all available rows immediately (similar to how `cat` behaves).
+ 
+ Examples: 
+ ``` bash
+ tctl workflow list --open --all | wc
+ ```
+ ``` bash
+ tctl workflow list --all | more
+ ```
+ 
+ - `grep` support: add a flag to output data as cards, where each row will follow format `{field name}: {field value}`
```

Comment:
I'm not sure I like the card output option.
What if we just support formatting with go templates like docker does? https://docs.docker.com/config/formatting/
I also like the `-q` option for docker commands which will only output IDs.

Reply:
i like that go-template: 
 - gives flexibility and power to users
 - makes it easy for us to develop (+less error prone)
 
On the bad side:
 - it's hardly user friendly since it will require majority of users to learn the template formatting if they want to output something similar to the proposed "card" format
 
What's your opinion on the following:
we add `--format` (or --output as you've mentioned [here](https://github.com/temporalio/proposals/pull/26/files#r653123514)) flag that accepts
 - table
 - json
 - card
 - <go-template> - everything that is not above 3 is considered go-template